mlflow/genai/judges/instructions_judge/__init__.py

```diff
@@ -0,0 +1,271 @@
+        """
+        super().__init__(name=name, **kwargs)
+
+        if not name or not isinstance(name, str):
+            raise MlflowException(
+                "name must be a non-empty string", error_code=INVALID_PARAMETER_VALUE
+            )
+        if not instructions or not isinstance(instructions, str):
+            raise MlflowException(
+                "instructions must be a non-empty string", error_code=INVALID_PARAMETER_VALUE
+            )
+
+        self._instructions = instructions
+        self._model = model or get_default_model()
+
+        self._instructions_prompt = PromptVersion(
+            name=name,
+            version=1,
+            template=instructions,
+        )
+
+        self._custom_template_variables = self._instructions_prompt.variables - set(
+            self._RESERVED_INSTRUCTION_TEMPLATE_VARIABLES
+        )
+
+        self._validate_model_format()
+        self._validate_instructions_template()
+
+    @property
+    def instructions(self) -> str:
+        """Get the instructions for this judge."""
+        return self._instructions
+
+    @property
+    def model(self) -> str:
+        """Get the model for this judge."""
+        return self._model
+
+    @property
+    def template_variables(self) -> set[str]:
+        """Get the template variables from the instructions."""
+        return self._instructions_prompt.variables
+
+    @property
+    def description(self) -> str:
+        """Get the description of this judge."""
+        header = f"Instructions-based judge: {self.name}"
+        return f"{header}\n\nInstructions:\n-------------\n\n{self._instructions}"
+
+    def __call__(
+        self,
+        *,
+        inputs: dict[str, Any] | None = None,
+        outputs: dict[str, Any] | None = None,
+        expectations: dict[str, Any] | None = None,
+        trace: Trace | None = None,
+    ) -> Any:
+        """
+        Evaluate the provided data using the judge's instructions.
+
+        NOTE: Evaluation supports either inputs, outputs, and/or expectations
+        or trace as input. A trace cannot be used with any other fields.
+
+        Args:
+            inputs: Input dictionary to evaluate. Cannot be used with 'trace'.
+            outputs: Output dictionary to evaluate. Cannot be used with 'trace'.
+            expectations: Expected outcomes or ground truth.
+            trace: Trace object for evaluation.
+
+        Returns:
+            Evaluation results
+
+        """
+        # TODO: update this to perform a mutual exclusion validiation check if both
+        # trace and inputs / outputs / expectations are submitted once trace template
+        # substitution is implemented
+        if trace is not None:
+            raise MlflowException(
+                "Trace evaluation is not supported in this version. "
+                "Please use 'inputs' and 'outputs' for evaluation.",
+                error_code=INVALID_PARAMETER_VALUE,
+            )
```

Comment:
For now, can we remove `trace` from the arguments in `__call__`? I can reintroduce later :)